WordPress/gutenberg
# Fix full-screen bottom sheet animations

## Related PRs

* https://github.com/wordpress-mobile/gutenberg-mobile/pull/3894

## Description
Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/3883. 

For iOS, the Link Picker now dismisses iOS keyboard to ensure smooth animation. Focus of the URL text input caused animation stutter while the iOS keyboard visibility toggled from hide to show to hide. To address this, the Android-specific keyboard dismiss logic is now applied to iOS as well.

For Android, animations have been reintroduced for full-screen bottom sheets. The `layout` conditional was removed in https://github.com/WordPress/gutenberg/pull/28632, as it is no longer utilized. However, its prior presence was universally disabling the block it guarded. Removing the always falsey conditional caused the block to always run and break the Android animation.

Specifically, when navigating from a full-screen bottom sheet to a non-full-screen bottom sheet, the `currentHeight` is always `1`. This results in setting the height without layout animations prior to this change.

## How has this been tested?

<details><summary>Link Picker</summary>

1. Insert a Paragraph block into the editor. 
2. Focus the Paragraph block. 
3. Tap the link icon button. 
4. Tap "Link to." 
5. Insert text into the URL field.
6. Tap "Add this link."
7. Tap "Link to." 
8. Tap "Cancel." 
9. ℹ️ **Expected:** The full-screen bottom animates closing without jitter. 

</details>
<details><summary>Button Background</summary>

1. Add or edit a Buttons block. 
2. Open the settings for an individual Button. 
3. Tap "background color."
4. Tap "Gradient."
5. Select a gradient.
6. Tap "Customize Gradient."
7. Tap "Back."
8. Tap "Solid."
9. Tap "Gradient."
10. ℹ️ **Expected:** The height of the bottom sheet is animated to the _correct height_ so all controls within the bottom sheet are visible. 

</details>

⚠️ Additional testing of various block settings bottom sheets to verify they function as expected is welcome. 

## Screenshots 

<details><summary>iOS Before</summary>

https://user-images.githubusercontent.com/438664/131565111-fe9f3560-5646-4aab-944d-e4573fb8e759.MP4

</details>
<details><summary>iOS After</summary>

https://user-images.githubusercontent.com/438664/131565133-9604b570-257b-4837-bd4c-4e83bec19ae6.MP4

</details>
<details><summary>Android Before</summary>

https://user-images.githubusercontent.com/438664/131565187-5bf800b6-c733-4ef2-946f-81c5533f9209.mp4

</details>
<details><summary>Android After</summary>

https://user-images.githubusercontent.com/438664/131565196-2c2fa5c8-1da1-4244-8007-a9f067b4c461.mp4

</details>

## Types of changes
Bug fix

## Checklist:
